src/Guzzle/MiddleWare/RequestLogger.php

```diff
@@ -0,0 +1,175 @@
+<?php
+
+namespace Acquia\ContentHubClient\Guzzle\Middleware;
+
+use Closure;
+use Exception;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * Class RequestLogger.
+ */
+class RequestLogger
+{
+
+    /**
+     * @var \Psr\Log\LoggerInterface
+     */
+    protected $logger;
+
+    /**
+     * @var \Psr\Http\Message\RequestInterface
+     */
+    protected $request;
+
+    /**
+     * @var \Psr\Http\Message\ResponseInterface
+     */
+    protected $response;
+
+    /**
+     * Decoded response body.
+     *
+     * @var mixed
+     */
+    protected $decodedResponseBody;
+
+    /**
+     * RequestLogger constructor.
+     *
+     * @param \Psr\Log\LoggerInterface $logger
+     */
+    public function __construct(LoggerInterface $logger)
+    {
```

Comment:
If you accept that Request and Response are inevitable parts of the class, then I expect to see them as constructor arguments as well. 
On a different note, decodedResponseBody could be defined private since it's a computed class member.

Reply:
Request and Response objects only available in promise function. We can't pass it into constructor.

Reply:
In that case, this class has more than one responsibility and should be broken down. I would rethink the design.

Reply:
Logic related to logging moved to separate class.